native/libcst/src/parser/grammar.rs

```diff
@@ -2991,3 +3214,272 @@ fn make_named_expr<'a>(name: Name<'a>, tok: TokenRef<'a>, expr: Expression<'a>)
+    match make_number(tok) {
+        e @ (Expression::Integer(_) | Expression::Float(_)) => Ok(e),
+        _ => Err("real number"),
+    }
+}
+
+fn ensure_imaginary_number(tok: TokenRef) -> GrammarResult<Expression> {
+    match make_number(tok) {
+        e @ Expression::Imaginary(_) => Ok(e),
+        _ => Err("imaginary number"),
+    }
+}
+
+fn make_tuple_pattern<'a>(
+    lpar: LeftParen<'a>,
+    patterns: Vec<StarrableMatchSequenceElement<'a>>,
+    rpar: RightParen<'a>,
+) -> MatchSequence<'a> {
+    MatchSequence::MatchTuple(MatchTuple {
+        patterns,
+        lpar: vec![lpar],
+        rpar: vec![rpar],
+    })
+}
+
+fn make_open_sequence_pattern<'a>(
+    first: StarrableMatchSequenceElement<'a>,
+    comma: Comma<'a>,
+    mut rest: Vec<StarrableMatchSequenceElement<'a>>,
+) -> Vec<StarrableMatchSequenceElement<'a>> {
+    rest.insert(0, first.with_comma(comma));
+    rest
+}
+
+fn make_match_sequence_element(value: MatchPattern) -> MatchSequenceElement {
+    MatchSequenceElement {
+        value,
+        comma: Default::default(),
+    }
+}
+
+fn make_match_star<'a>(star_tok: TokenRef<'a>, name: Option<Name<'a>>) -> MatchStar<'a> {
+    MatchStar {
+        name,
+        comma: Default::default(),
+        whitespace_before_name: Default::default(),
+        star_tok,
+    }
+}
+
+fn make_match_mapping<'a>(
+    lbrace: LeftCurlyBrace<'a>,
+    mut elements: Vec<MatchMappingElement<'a>>,
+    el_comma: Option<Comma<'a>>,
+    star_tok: Option<TokenRef<'a>>,
+    rest: Option<Name<'a>>,
+    trailing_comma: Option<Comma<'a>>,
+    rbrace: RightCurlyBrace<'a>,
+) -> MatchPattern<'a> {
+    if let Some(c) = el_comma {
+        if let Some(el) = elements.pop() {
+            elements.push(el.with_comma(c));
+        }
+        // TODO: else raise error
+    }
+    MatchPattern::Mapping(MatchMapping {
+        elements,
+        rest,
+        trailing_comma,
+        lbrace,
+        rbrace,
+        lpar: Default::default(),
+        rpar: Default::default(),
+        whitespace_before_rest: Default::default(),
+        star_tok,
+    })
+}
+
+fn make_match_mapping_element<'a>(
+    key: Expression<'a>,
+    colon_tok: TokenRef<'a>,
+    pattern: MatchPattern<'a>,
+) -> MatchMappingElement<'a> {
+    MatchMappingElement {
+        key,
+        pattern,
+        comma: Default::default(),
+        whitespace_before_colon: Default::default(),
+        whitespace_after_colon: Default::default(),
+        colon_tok,
+    }
+}
+
+fn make_class_pattern<'a>(
+    cls: NameOrAttribute<'a>,
+    lpar_tok: TokenRef<'a>,
+    mut patterns: Vec<MatchSequenceElement<'a>>,
+    pat_comma: Option<Comma<'a>>,
+    mut kwds: Vec<MatchKeywordElement<'a>>,
+    kwd_comma: Option<Comma<'a>>,
+    rpar_tok: TokenRef<'a>,
+) -> MatchPattern<'a> {
+    if let Some(c) = pat_comma {
+        if let Some(el) = patterns.pop() {
+            patterns.push(el.with_comma(c));
+        }
+        // TODO: else raise error
```

Comment:
do we need to handle these TODOs before merging? There are several of them both above and below this one

Reply:
Not really. There's no good way of erroring out the parsing apart from `panic!()` at the moment, and these cases shouldn't be happening unless there's a bug in the grammar.